website/project/model.py

```diff
@@ -2510,6 +2544,101 @@ def serialize(self, auth=None):
             'is_registration': self.is_registration,
         }
 
+    def _initiate_retraction(self, user, justification=None):
+        """Initiates the retraction process for a registration
+        :param user: User who initiated the retraction
+        :param justification: Justification, if given, for retraction
+        """
+
+        retraction = Retraction()
+        retraction.initiated_by = user
+        if justification:
+            retraction.justification = justification
+        retraction.state = 'pending'
+
+        admins = [contrib for contrib in self.contributors if self.has_permission(contrib, 'admin')]
+
+        approval_state = {}
+        # Create approve/disapprove tokens
+        for admin in admins:
+            approval_state[admin._id] = {
+                'approval_token': security.random_string(30),
+                'disapproval_token': security.random_string(30),
+                'has_approved': False
+            }
+
+        retraction.approval_state = approval_state
+        return retraction
+
+    def retract_registration(self, user, justification=None):
+        """Retract public registration. Instantiate new Retraction object
+        and associate it with the respective registration.
+        """
+
+        if not self.is_public or not self.is_registration:
+            raise NodeStateError('Cannot retract private node or non-registration')
+
+        retraction = self._initiate_retraction(user, justification)
```

Comment:
This is just my opinion-- consider doing ```retraction.save()``` in #initiate_retraction for better encapsulation. 

Maybe even make the signature of #retract_registration: 
```def retract_registration(self, user, justification=None, save=True)```
and doing a:
``` 
if save:
    retraction.save()
```